### ISSUE TITLE
e2e failing on 1.1: Kubectl client Guestbook application should create and stop a working application [Conformance]

### ISSUE THREAD

**Issue by jlowdermilk** [kubernetes/kubernetes#25590]
The Guestbook test started failing on May 10th on the 1.1 and 1.0 branches. It looks like the failure is caused by inability to hit the Guestbook's frontend service via the proxy. See [kubernetes-e2e-gce-release-1.1/2523](https://console.cloud.google.com/storage/browser/kubernetes-jenkins/logs/kubernetes-e2e-gce-release-1.1/2523) for example failure.

I ran the guestbook example on a 1.1 cluster, and it appears to run correctly but I can't figure out out to hit the frontend via proxy.
```
kubectl version                                                  
Client Version: version.Info{Major:"1", Minor:"2", GitVersion:"v1.2.3", GitCommit:"882d296a99218da8f6b2a340eb0e81c69e66ecc7", GitTreeState:"clean"}
Server Version: version.Info{Major:"1", Minor:"1+", GitVersion:"v1.1.9-beta.0.31+ccd48523af23a0", GitCommit:"ccd48523af23a09cd05bbfe6cf82efe69057b08b", GitTreeState:"clean"}

kubectl get pods
NAME                 READY     STATUS    RESTARTS   AGE
frontend-mx6xo       1/1       Running   0          34m
frontend-sxqj0       1/1       Running   0          34m
frontend-yrakm       1/1       Running   0          34m
redis-master-bhxkr   1/1       Running   0          34m
redis-slave-4jsnr    1/1       Running   0          34m
redis-slave-mvnds    1/1       Running   0          34m

kubectl proxy &
[2] 20375
Starting to serve on 127.0.0.1:8001

curl http://127.0.0.1:8001/api/v1/proxy/namespaces/default/services/frontend/guestbook.php
Error: 'dial tcp 10.245.0.7:80: i/o timeout'
Trying to reach: 'http://10.245.0.7:80/guestbook.php'%                                                                                                                                                                                

kubectl get endpoints
NAME           ENDPOINTS                                   AGE
frontend       10.245.0.7:80,10.245.1.8:80,10.245.2.8:80   36m
kubernetes     10.240.0.2:443                              2h
redis-master   10.245.2.9:6379                             36m
redis-slave    10.245.0.8:6379,10.245.2.10:6379            36m
```

I'm not 100% on path for hitting via proxy but that endpoint looks correct. I can see that the frontend is running in the pod

```
kubectl exec frontend-mx6xo -- curl localhost:80                                          
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0<html ng-app="redis">
  <head>
    <title>Guestbook</title>
    <link rel="stylesheet" href="//netdna.bootstrapcdn.com/bootstrap/3.1.1/css/bootstrap.min.css">
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.2.12/angular.min.js"></script>
    <script src="controllers.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/angular-ui-bootstrap/0.13.0/ui-bootstrap-tpls.js"></script>
  </head>
  <body ng-controller="RedisCtrl">
    <div style="width: 50%; margin-left: 20px">
      <h2>Guestbook</h2>
    <form>
    <fieldset>
    <input ng-model="msg" placeholder="Messages" class="form-control" type="text" name="input"><br>
    <button type="button" class="btn btn-primary" ng-click="controller.onRedis()">Submit</button>
    </fieldset>
    </form>
    <div>
      <div ng-repeat="msg in messages track by $index">
        {{msg}}
      </div>
    </div>
    </div>
  </body>
</html>
100   921  100   921    0     0   179k      0 --:--:-- --:--:-- --:--:--  224k
```

cc @lavalamp @krousey

**Comment by lavalamp** [comment]
You're using a new client w/ an old server? maybe there's a compatibility
bug?

**Comment by krousey** [comment]
For a quick sanity check, what does `curl http://127.0.0.1:8001/api/v1/proxy/namespaces/default/services/frontend/` do?

**Comment by jlowdermilk** [comment]
Same thing with 1.1 client:
```
cluster/kubectl.sh version           
Client Version: version.Info{Major:"1", Minor:"1+", GitVersion:"v1.1.9-beta.0.31+ccd48523af23a0", GitCommit:"ccd48523af23a09cd05bbfe6cf82efe69057b08b", GitTreeState:"clean"}
Server Version: version.Info{Major:"1", Minor:"1+", GitVersion:"v1.1.9-beta.0.31+ccd48523af23a0", GitCommit:"ccd48523af23a09cd05bbfe6cf82efe69057b08b", GitTreeState:"clean"}

cluster/kubectl.sh proxy &
[2] 15692
Starting to serve on 127.0.0.1:8001

curl http://127.0.0.1:8001/api/v1/proxy/namespaces/default/services/frontend           
curl http://127.0.0.1:8001/api/v1/proxy/namespaces/default/services/frontend/
Error: 'dial tcp 10.245.2.8:80: i/o timeout'
Trying to reach: 'http://10.245.2.8:80/'%  
```
Interestingly, `curl http://127.0.0.1:8001/api/v1/proxy/namespaces/default/services/frontend` exits 0 with no output.

**Comment by jlowdermilk** [comment]
The plot thickens. I now suspect this to be broken at head and the test just isn't catching it. From [kubernetes-e2e-gce/16839](https://console.cloud.google.com/storage/browser/kubernetes-jenkins/logs/kubernetes-e2e-gce/16839):
```
16:31:24 May 13 16:29:07.602: INFO: stderr: ""
16:31:24 May 13 16:29:07.602: INFO: stdout: "service \"redis-slave\" created\n"
16:31:24 STEP: validating guestbook app
16:31:24 May 13 16:29:07.602: INFO: Waiting for frontend to serve content.
16:31:24 May 13 16:29:07.605: INFO: Failed to get response from guestbook. err: endpoints "frontend" not found, response: 
16:31:24 May 13 16:29:12.615: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:29:17.621: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:29:22.625: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:29:27.629: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:29:32.658: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:29:37.662: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:29:42.669: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:29:47.674: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:29:52.677: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:29:57.684: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:30:02.691: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:30:07.698: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:30:12.706: INFO: Failed to get response from guestbook. err: no endpoints available for service "frontend", response: 
16:31:24 May 13 16:30:22.739: INFO: Failed to get response from guestbook. err: <nil>, response: <br />
16:31:24 <b>Fatal error</b>:  Uncaught exception 'Predis\Connection\ConnectionException' with message 'Connection timed out [tcp://redis-slave:6379]' in /usr/local/lib/php/Predis/Connection/AbstractConnection.php:168
16:31:24 Stack trace:
16:31:24 #0 /usr/local/lib/php/Predis/Connection/StreamConnection.php(97): Predis\Connection\AbstractConnection-&gt;onConnectionError('Connection time...', 110)
16:31:24 #1 /usr/local/lib/php/Predis/Connection/StreamConnection.php(58): Predis\Connection\StreamConnection-&gt;tcpStreamInitializer(Object(Predis\Connection\Parameters))
16:31:24 #2 /usr/local/lib/php/Predis/Connection/AbstractConnection.php(94): Predis\Connection\StreamConnection-&gt;createResource()
16:31:24 #3 /usr/local/lib/php/Predis/Connection/StreamConnection.php(158): Predis\Connection\AbstractConnection-&gt;connect()
16:31:24 #4 /usr/local/lib/php/Predis/Connection/AbstractConnection.php(193): Predis\Connection\StreamConnection-&gt;connect()
16:31:24 #5 /usr/local/lib/php/Predis/Connection/StreamConnection.php(184): Predis\Connection\AbstractConnection-&gt;getResource()
16:31:24 #6 /usr/local/lib/php/Predis/Connection/Stre in <b>/usr/local/lib/php/Predis/Connection/AbstractConnection.php</b> on line <b>168</b><br />
16:31:24 
16:31:24 May 13 16:30:27.775: INFO: Trying to add a new entry to the guestbook.
16:31:24 May 13 16:30:27.810: INFO: Verifying that added entry can be retrieved.
```

This is the same error that I see when I try to curl from inside the frontend container on 1.1:
```
kubectl exec frontend-0zhre -- curl 'localhost:80/guestbook.php?cmd=get&key=messages'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:30 --:--:--     0<br />
<b>Fatal error</b>:  Uncaught exception 'Predis\Connection\ConnectionException' with message 'php_network_getaddresses: getaddrinfo failed: Name or service not known [tcp://redis-slave:6379]' in /usr/local/lib/php/Predis/Connection/AbstractConnection.php:168
Stack trace:
#0 /usr/local/lib/php/Predis/Connection/StreamConnection.php(97): Predis\Connection\AbstractConnection-&gt;onConnectionError('php_network_get...', 0)
#1 /usr/local/lib/php/Predis/Connection/StreamConnection.php(58): Predis\Connection\StreamConnection-&gt;tcpStreamInitializer(Object(Predis\Connection\Parameters))
#2 /usr/local/lib/php/Predis/Connection/AbstractConnection.php(94): Predis\Connection\StreamConnection-&gt;createResource()
#3 /usr/local/lib/php/Predis/Connection/StreamConnection.php(158): Predis\Connection\AbstractConnection-&gt;connect()
#4 /usr/local/lib/php/Predis/Connection/AbstractConnection.php(193): Predis\Connection\StreamConnection-&gt;connect()
#5 /usr/local/lib/php/Predis/Connection/StreamConnection.php(184): Predis\Connection\AbstractConnection-&gt;getResour in <b>/usr/local/lib/php/Predis/Connection/AbstractConnection.php</b> on line <b>168</b><br />
100  1166  100  1166    0     0     38      0  0:00:30  0:00:30 --:--:--   273
```

**Comment by bgrant0607** [comment]
cc @kubernetes/examples

**Comment by bgrant0607** [comment]
cc @ncdc

**Comment by jlowdermilk** [comment]
Fixed by #27139
